kf_lib_data_ingest/target_api_plugins/kids_first_dataservice.py

```diff
@@ -668,9 +678,16 @@ def query_target_ids(cls, host, key_components):
 
     @classmethod
     def build_entity(cls, record, get_target_id_from_record):
+        def external_id(record):
```

Comment:
Could be in another PR, sorry 😬 , but I snuck these external_id additions into each of the many-to-many association tables. I like seeing the external_id field filled in for these in Data Service because it gives me some idea of which two things are related from a source data perspective. Pretty much for debugging purposes or if I need to do a delete or bulk delete. For example seeing `external_id:  Schiffman-SS01-file_abc12345.bam` gives me an idea of which specimen and file are related in this BG record.

Reply:
I kinda think these should be using a string version of otherclass.get_key_components(record, get_target_id_from_record) no? biospecimen ID alone doesn't uniquely identify a specimen

Reply:
Mm hmm yea I think you're right

Reply:
Oh no, nevermind. That results in values like this for `external_id: PT_TP0FPEXC-BS0-BS0-SD_ME0WME0W-GF0`. I want the values from the source data

Reply:
Hmm. What about putting participant_id in build_entity instead of key components on biospecimen and replace it with study_id. Then you can just remove the study_id field here before concatenating values.